tools/src/main/java/org/apache/kafka/trogdor/fault/DegradedNetworkFaultWorker.java

```diff
@@ -96,12 +96,21 @@ public void stop(Platform platform) throws Exception {
         return devices;
     }
 
-    private void enableTrafficControl(Platform platform, String networkDevice, int delayMs) throws IOException {
+    private void enableTrafficControl(Platform platform, String networkDevice, int delayMs, int rateLimitKbps) throws IOException {
         int deviationMs = Math.max(1, (int) Math.sqrt(delayMs));
         platform.runCommand(new String[] {
-            "sudo", "tc", "qdisc", "add", "dev", networkDevice, "root", "netem",
-            "delay", String.format("%dms", delayMs), String.format("%dms", deviationMs), "distribution", "normal"
+            "sudo", "tc", "qdisc", "add", "dev", networkDevice, "handle", "1:0",
+            "netem", "delay", String.format("%dms", delayMs), String.format("%dms", deviationMs), "distribution", "paretonormal"
         });
+
+        if (rateLimitKbps > 0) {
+            int maxLatency = delayMs * delayMs;
+            platform.runCommand(new String[]{
+                // tc qdisc add dev eth0 parent 1:1 handle 10: tbf rate 256kbit buffer 1600 limit 3000
+                "tc", "qdisc", "add", "dev", "eth0", "parent", "1:1", "handle", "10:",
```

Comment:
few comments on this line:
1. use `networkDevice` instead of `eth0`. 
2. we are not using `sudo` here, but use in when adding `delay` above.
3. We are using `parent` and `handle` that seem different from command above. I think these values matter. I need to re-read the doc I wrote, but I think these specify the nodes in the device tree where rate will be limited.

Reply:
fixed